test/upgrade.js

```diff
@@ -0,0 +1,106 @@
+const chai = require("chai");
+const { ethers, upgrades } = require("hardhat");
+const {solidity} = require("ethereum-waffle");
+chai.use(solidity);
+const {expect, assert} = chai;
+
+describe("Contract Upgrades", function () {
+  let Token;
+  let arrowToken;
+  let TokenV2;
+  let arrowTokenV2;
+  let owner;
+  let signer1;
+  let amountTransferredToSigner1;
+
+  beforeEach(async function () {
+    Token = await ethers.getContractFactory("ArrowToken");
+    [owner, signer1] = await ethers.getSigners();
+    let name = "Arrow Token";
+    let symbol = "ARROW";
+    arrowToken = await upgrades.deployProxy(
+      Token,
+      [ethers.BigNumber.from(1_000_000), name, symbol],
+      { kind: "uups" }
+    );
+    await arrowToken.deployed();
+    
+    // Transfer all of owner's tokens to signer1
+    let senderBalance = ethers.BigNumber.from(await arrowToken.balanceOf(owner.address));
+    amountTransferredToSigner1 = senderBalance;
+    await arrowToken.transfer(signer1.address, amountTransferredToSigner1);
+    expect(await arrowToken.balanceOf(owner.address)).to.equal(
+      senderBalance - amountTransferredToSigner1
+    );
```

Comment:
Why are we doing this in the `beforeEach`? It seems unnecessary for the vast number of tests.